allensdk/brain_observatory/filtered_running_speed/filtered_running_speed.py

```diff
@@ -0,0 +1,370 @@
+
+        return sync_data.get_edges(
+            "rising", SyncDataset.FRAME_KEYS, units="seconds"
+        )
+
+    def _get_stimulus_starts_and_ends(self):
+        """
+        Get the start and stop frame indexes for each stimulus
+        """
+
+        (
+            behavior_frame_count,
+            mapping_frame_count,
+            replay_frames_count
+        ) = self._get_frame_counts()
+
+        behavior_start = FilteredRunningSpeed.START_FRAME
+        behavior_end = behavior_frame_count
+
+        mapping_start = behavior_end
+        mapping_end = mapping_start + mapping_frame_count
+
+        replay_start = mapping_end
+        replay_end = replay_start + replay_frames_count
+
+        return (
+            behavior_start,
+            behavior_end,
+            mapping_start,
+            mapping_end,
+            replay_start,
+            replay_end
+        )
+
+    def _merge_dx_data(
+        self,
+        mapping_velocities,
+        behavior_velocities,
+        replay_velocities,
+        frame_times
+    ):
+        """
+        Concatenate all of the running speed data
+
+        Parameters
+        ----------
+        mapping_velocities: pandas.core.frame.DataFrame
+            Velocity data from mapping stimulus
+        behavior_velocities: pandas.core.frame.DataFrame
+           Velocity data from behavior stimulus
+        replay_velocities: pandas.core.frame.DataFrame
+            Velocity data from replay stimulus
+        frame_times: numpy.ndarray
+            list of the vsync times
+
+        Returns
+        -------
+        list[pd.DataFrame, pd.DataFrame]
+            concatenated velocity data, raw data
+        """
+
+        speed = np.concatenate(
+            (
+                behavior_velocities['speed'],
+                mapping_velocities['speed'],
+                replay_velocities['speed']),
+            axis=None
+        )
+
+        dx = np.concatenate(
+            (
+                behavior_velocities['dx'],
+                mapping_velocities['dx'],
+                replay_velocities['dx']),
+            axis=None
+        )
+
+        vsig = np.concatenate(
+            (
+                behavior_velocities['v_sig'],
+                mapping_velocities['v_sig'],
+                replay_velocities['v_sig']),
+            axis=None
+        )
+
+        vin = np.concatenate(
+            (
+                behavior_velocities['v_in'],
+                mapping_velocities['v_in'],
+                replay_velocities['v_in']),
+            axis=None
+        )
+
+        frame_indexes = list(
+            range(FilteredRunningSpeed.START_FRAME, len(frame_times))
+        )
+
+        velocities = pd.DataFrame(
+            {
+                "velocity": speed,
+                "net_rotation": dx,
+                "frame_indexes": frame_indexes,
+                "frame_time": frame_times
+            }
+        )
+
+        # due to an acquisition bug (the buffer of raw orientations
```

Comment:
This comment is duplicated [here](https://github.com/AllenInstitute/AllenSDK/blob/master/allensdk/brain_observatory/extract_running_speed/__main__.py#L79). Is there any way to reduce duplication?

Reply:
This line is duplicated:  df = df[~(np.isclose(df["net_rotation"], 0.0))] but I do not think that much else is. I could put this line in a separate util object and call it in both places. I'm hesitant to refactor this code that you linked to though.